launchlib/memory.go

```diff
@@ -54,18 +63,58 @@ func (c CGroupMemoryLimit) MemoryLimitInBytes() (uint64, error) {
 		return 0, errors.Wrap(err, "failed to get memory cgroup path")
 	}
 
-	memLimitFilepath := filepath.Join(memoryCGroupPath, memLimitName)
-	memLimitFile, err := c.fs.Open(convertToFSPath(memLimitFilepath))
+	isCGroupV2, err := IsCGroupV2(c.fs)
```

Comment:
We're checking `isCGroupV2` in `NewCGroupMemoryLimit` as well. What are your thoughts on extending `CGroupMemoryLimit` with a field for `version` or `boolean isCgroupV2`? Can avoid a double-check I think.